### ISSUE TITLE
move out nlp components

### ISSUE THREAD

**Issue by imgarylai** [elitcloud/elit#33]
This elit project will only be the sdk for deployment to the cloud. Since we have many overlapping with gluon nlp, it might be a better way for us that we contribute nlp model to there. Since then, we don't have to care about having a unified api interface for each nlp component. 

In the future, developers don't have to worry about how should implement their component. The only thing they should care about is the `decode` function. 

I have created a snapshot on https://github.com/elitcloud/elit/tree/nlp for this move out process. I'm gonna move out every components after this commit 9e4f1855b8aa06f51b49aa2b62f97e2df8893af8. 

@hankcs would you please move your parser to https://github.com/elitcloud/parser? 

Please leave your ideas/thoughts here for further discussion. 

Best

**Comment by hankcs** [comment]
Hum, changes in package again. Seems that I need to re-train parser model again.

**Comment by imgarylai** [comment]
Is it because of the path? Can it be solved by changing the config path?

**Comment by hankcs** [comment]
Some information is saved via `pickle.save`. Maybe we've spent lots of resource on adjusting package structure.

**Comment by hankcs** [comment]
Maybe `json` would be a better approach than `pickle`.

**Comment by imgarylai** [comment]
Yes, that is the part of question we really want to solve. If every component is developed stand alone, the developer can just do whatever they want in their way. Also, the elit api might change, and I don't want the change in this project affects any comp at all. That's why we want to move out again. 

@jdchoi77 Please review this issue. 

I had this issue with pickle before. This can be solved by read it and replace the config by changing the data object. 

Also, json is easier to use, I like json. Either way works fine.

**Comment by Jeevi10** [comment]
I installed and ran below  code
from elit.structure import Document, Sentence, TOK, POS
from elit.component import SDPBiaffineParser

tokens = ['John', 'who', 'I', 'wanted', 'to', 'meet', 'was', 'smart']
postags = ['NNP', 'IN', 'WP', 'PRP', 'VBD', 'DT', 'NN', 'VBD', 'JJ'] 
doc = Document()
doc.add_sentence(Sentence({TOK: tokens, POS: postags}))
#print(doc)
sdp = SDPBiaffineParser()
sdp.decode(doc)
print(doc.sentences[0])

and it throws error as shown below. 



AttributeError                            Traceback (most recent call last)
<ipython-input-78-fb62d2a9863d> in <module>
      8 #print(doc)
      9 sdp = SDPBiaffineParser()
---> 10 sdp.decode(doc)
     11 print(doc.sentences[0])

~/anaconda3/lib/python3.7/site-packages/elit/component/sdp/sdp_parser.py in decode(self, docs, num_buckets_test, test_batch_size, **kwargs)
    119         :return: docs
    120         """
--> 121         vocab = self._parser._vocab
    122         for d in docs:
    123             for s in d:

AttributeError: 'NoneType' object has no attribute '_vocab'
